lib/auth/github_authenticator.rb

```diff
@@ -113,4 +117,15 @@ def register_middleware(omniauth)
            },
            scope: "user:email"
   end
+
+  protected
+
+  def retrieve_avatar(user, data)
+    return unless user
+    return if user.user_avatar&.custom_upload_id.present?
+
+    if (avatar_url = data[:image]).present?
```

Comment:
I would move this condition to the first line of this method and just `return` if there isn't `avatar_url`